FROM_PROTO.md

```diff
@@ -30,6 +32,98 @@ substreams-sink-sql from-proto <dsn> <manifest> [output-module]
 - `--no-constraints`: Skip adding database constraints (useful for fast initial imports)
 - `--block-batch-size`: Number of blocks to process at a time (default: 25)
 
+## DSN Format
+
+The Data Source Name (DSN) specifies how to connect to your database. The format varies by database type:
+
+### PostgreSQL DSN Format
+
+```
+postgres://[username[:password]@]host[:port]/database[?param1=value1&param2=value2]
+```
+
+**Examples:**
+```bash
+# Basic connection
+postgres://localhost:5432/postgres
+
+# With authentication (replace with actual credentials)
+postgres://[username]:[password]@localhost:5432/mydb
+
+# With SSL disabled and custom schema
+postgres://localhost:5432/postgres?sslmode=disable&schemaName=orders
+
+# Production example with SSL (replace placeholders)
+postgres://[username]:[password]@[hostname]:5432/analytics?sslmode=require
+```
+
+**Common PostgreSQL Parameters:**
+- `sslmode`: SSL connection mode (`disable`, `require`, `verify-ca`, `verify-full`)
+- `schemaName`: Target schema name (defaults to `public`)
+- `connect_timeout`: Connection timeout in seconds
+- `application_name`: Application name for connection tracking
+
+### ClickHouse DSN Format
+
+```
+clickhouse://[username[:password]@]host[:port]/database[?param1=value1&param2=value2]
+```
+
+**Examples:**
+```bash
+# Basic connection (default user, no password)
+clickhouse://127.0.0.1:9000/order?secure=false
+
+# With authentication (replace with actual credentials)
+clickhouse://[username]:[password]@localhost:9000/analytics
+
+# Secure connection (replace placeholders)
+clickhouse://[username]:[password]@[hostname]:9440/mydb?secure=true
+
+# With custom settings
+clickhouse://localhost:9000/analytics?secure=false&compress=true&debug=true
+```
+
+**Common ClickHouse Parameters:**
+- `secure`: Use TLS encryption (`true` or `false`)
+- `compress`: Enable compression (`true` or `false`)
+- `debug`: Enable debug logging (`true` or `false`)
+- `dial_timeout`: Connection timeout
+- `max_execution_time`: Query execution timeout
+
+### Environment Variable Usage
+
+For security, avoid hardcoding credentials in commands. Use environment variables:
+
+```bash
+# Set DSN as environment variable (replace with actual credentials)
+export DSN="postgres://[username]:[password]@localhost:5432/mydb?sslmode=disable"
+
+# Use in command
+substreams-sink-sql from-proto $DSN substreams.yaml
+```
+
+### Database Setup Examples
+
+**PostgreSQL with Docker:**
+```bash
+docker run --name postgres \
+  -e POSTGRES_PASSWORD=password \
+  -e POSTGRES_DB=analytics \
+  -p 5432:5432 -d postgres:13
+
+export DSN="postgres://postgres:password@localhost:5432/analytics?sslmode=disable"
+```
+
+**ClickHouse with Docker:**
+```bash
+docker run --name clickhouse \
+  -p 9000:9000 -p 8123:8123 \
+  -d clickhouse/clickhouse-server
+
+export DSN="clickhouse://127.0.0.1:9000/default?secure=false"
+```
+
 ## Step-by-Step Workflow
 
 ### Step 1: Create Your Protocol Buffer Definition
```

Comment:
Reuse the transfer example from https://github.com/streamingfast/substreams-sink-clickhouse-showcase instead of creating your own